pkg/config/polling_manager.go

```diff
@@ -89,21 +97,40 @@ func InitialDatafile(datafile []byte) OptionFunc {
 func (cm *PollingProjectConfigManager) SyncConfig(datafile []byte) {
 	var e error
 	var code int
+	var respHeaders http.Header
 
 	closeMutex := func(e error) {
 		cm.err = e
 		cm.configLock.Unlock()
 	}
 
 	if len(datafile) == 0 {
-		datafile, code, e = cm.requester.Get()
+		if cm.lastModified != "" {
+			lastModifiedHeader := utils.Header{Name: ModifiedSince, Value: cm.lastModified}
+			datafile, respHeaders, code, e = cm.requester.Get(lastModifiedHeader)
+		} else {
+			datafile, respHeaders, code, e = cm.requester.Get()
+		}
 
 		if e != nil {
 			cmLogger.Error(fmt.Sprintf("request returned with http code=%d", code), e)
 			cm.configLock.Lock()
 			closeMutex(e)
 			return
 		}
+
+		if code == 304 {
+			cmLogger.Debug("The datafile was not modified and won't be downloaded again")
```

Comment:
if we want to go this way, I am not sure if the logic : 
```
if projectConfig.GetRevision() == previousRevision {
		cmLogger.Debug(fmt.Sprintf("No datafile updates. Current revision number: %s", cm.projectConfig.GetRevision()))
		closeMutex(nil)
		return
	}
``` 
is still needed. It is saying the same thing. I am not even sure if there is a real case scenario for that logic to be executed.

Reply:
What about if anyone is providing URL template (server) which doesn't provide status 304? but keeps the datafile same?

Reply:
are you saying that somebody would try to reimplement requester ? Our default requester is using http libs and I think you will always get Last Modified header. @mikeng13 , what do you think ?

Reply:
no am saying, like in FSC we use our datafile server, if anyone using like that for testing their own stuff then it may cause loading everytime.